vtex-sites/storeframework.store
# feat: Integrate Incentives with CMS

## What's the purpose of this pull request?

This PR aims to integrate `Incentives` with the CMS, by adding its properties.

## How does it work?

In the file `sections.json` we can define all the editable properties for each sections. Also it was necessary to update the componentes `Incentives`, `IncentivesHeader` and `IncentivesFooter` to receive/pass the CMS data properly. 

Final Result in my [developer workspace](https://israel--storeframework.myvtex.com/admin/new-cms/faststore/home/edit/ad2fd81d-a53c-4281-8d01-a4fc2f274db3/):

https://user-images.githubusercontent.com/10616243/162507648-1ffceda9-ed6c-4b59-8ab5-2c0f8f022bf0.mp4

## How to test it?

- Set your workspace to `israel` with the command `vtex use israel`
- Sync your project with the CMS watch `vtex cms sync --watch`
- Open the CMS to edit the data in the [developer workspace](https://israel--storeframework.myvtex.com/admin/new-cms/faststore/home/edit/ad2fd81d-a53c-4281-8d01-a4fc2f274db3/)
- Run your project with `yarn develop` to see the data you've edited in the CMS (P.S.:  Every time I edit things in CMS I had to rerun my local project to get the new data)

## Disclaimer

At the moment, there are some _ghost_ 👻  sections showing up in this dev workspace: 
<img width="1096" alt="image" src="https://user-images.githubusercontent.com/10616243/162509983-c5aee4f6-6c2c-4b6a-a46c-f0804f7c2609.png">

This should be caused possibly by another PR that was synced with the `master` workspace, but this should be ok once everything is merged. 

## References

- Related [Jira Task](https://vtex-dev.atlassian.net/browse/FSSS-261)
- [Integrating Store front with CMS](https://faststore.dev/tutorials/cms/0)
- Base.store [PR](https://github.com/vtex-sites/base.store/pull/474) that syncs Incentive component updates

Special thanks to @lmarinho e @mateuspontes for all the support 🤗 

## Checklist

<em>You may erase this after checking them all ;)</em>

- [x] Update sections.json
- [x] Update Incentives componentes to get/pass CMS data
